src/libstd/primitive_docs.rs

```diff
@@ -891,9 +897,10 @@ mod prim_usize { }
 /// A reference represents a borrow of some owned value. You can get one by using the `&` or `&mut`
 /// operators on a value, or by using a `ref` or `ref mut` pattern.
 ///
-/// For those familiar with pointers, a reference is just a pointer that is assumed to not be null.
-/// In fact, `Option<&T>` has the same memory representation as a nullable pointer, and can be
-/// passed across FFI boundaries as such.
+/// For those familiar with pointers, a reference is just a pointer that is assumed to be
+/// aligned, not null, and pointing to valid (initialized) memory.
```

Comment:
I think this is too vague. What does "initialized" mean here ? A `&MaybeUninit<bool>` can point to an allocation that was never "touched" or "initialized" in any way, because that uninitialized memory is "valid" for `MaybeUninit<bool>`.

```suggestion
/// aligned, not null, and pointing to memory containing a valid value of `T` - for example, `&bool` can only point to an allocation containing the integer values `1` (`true`) or `0` (`false`), but the behavior of creating a `&bool` that points to an allocation containing the value `3` is undefined.
```

Reply:
I agree this can be confusing.

Does this mean we should update https://doc.rust-lang.org/nightly/core/mem/union.MaybeUninit.html#method.assume_init as well?